compile.c

```diff
@@ -4589,6 +4593,19 @@ keyword_node_p(const NODE *const node)
     return nd_type_p(node, NODE_HASH) && (RNODE_HASH(node)->nd_brace & HASH_BRACE) != HASH_BRACE;
 }
 
+static VALUE
+get_symbol_value(const NODE *node)
+{
+    switch (nd_type(node)) {
+      case NODE_LIT:
+        return RNODE_LIT(node)->nd_lit;
+      case NODE_SYM:
+        return rb_node_sym_string_val(node);
+      default:
+        rb_bug("unexpected node: %s", ruby_node_name(nd_type(node)));
```

Comment:
`UNKNOWN_NODE` would be preferable.

Reply:
The change set will be like below, does it make sense?

```diff
diff --git a/compile.c b/compile.c
index 6e05b7277b..feaf65b7c9 100644
--- a/compile.c
+++ b/compile.c
@@ -4594,7 +4594,7 @@ keyword_node_p(const NODE *const node)
 }

 static VALUE
-get_symbol_value(const NODE *node)
+get_symbol_value(rb_iseq_t *iseq, const NODE *node)
 {
     switch (nd_type(node)) {
       case NODE_LIT:
@@ -4602,19 +4602,19 @@ get_symbol_value(const NODE *node)
       case NODE_SYM:
         return rb_node_sym_string_val(node);
       default:
-        rb_bug("unexpected node: %s", ruby_node_name(nd_type(node)));
+        UNKNOWN_NODE("get_symbol_value", node, Qnil);
     }
 }

 static VALUE
-node_hash_unique_key_index(rb_node_hash_t *node_hash, int *count_ptr)
+node_hash_unique_key_index(rb_iseq_t *iseq, rb_node_hash_t *node_hash, int *count_ptr)
 {
     NODE *node = node_hash->nd_head;
     VALUE hash = rb_hash_new();
     VALUE ary = rb_ary_new();

     for (int i = 0; node != NULL; i++, node = RNODE_LIST(RNODE_LIST(node)->nd_next)->nd_next) {
-        VALUE key = get_symbol_value(RNODE_LIST(node)->nd_head);
+        VALUE key = get_symbol_value(iseq, RNODE_LIST(node)->nd_head);
         VALUE idx = rb_hash_aref(hash, key);
         if (!NIL_P(idx)) {
             rb_ary_store(ary, FIX2INT(idx), Qfalse);
@@ -4674,7 +4674,7 @@ compile_keyword_arg(rb_iseq_t *iseq, LINK_ANCHOR *const ret,
         node = RNODE_HASH(root_node)->nd_head;
         {
             int len = 0;
-            VALUE key_index = node_hash_unique_key_index(RNODE_HASH(root_node), &len);
+            VALUE key_index = node_hash_unique_key_index(iseq, RNODE_HASH(root_node), &len);
             struct rb_callinfo_kwarg *kw_arg =
                 rb_xmalloc_mul_add(len, sizeof(VALUE), sizeof(struct rb_callinfo_kwarg));
             VALUE *keywords = kw_arg->keywords;
@@ -4690,7 +4690,7 @@ compile_keyword_arg(rb_iseq_t *iseq, LINK_ANCHOR *const ret,
                 const NODE *val_node = RNODE_LIST(RNODE_LIST(node)->nd_next)->nd_head;
                 int popped = TRUE;
                 if (rb_ary_entry(key_index, i)) {
-                    keywords[j] = get_symbol_value(key_node);
+                    keywords[j] = get_symbol_value(iseq, key_node);
                     j++;
                     popped = FALSE;
                 }
@@ -7099,7 +7099,7 @@ iseq_compile_pattern_each(rb_iseq_t *iseq, LINK_ANCHOR *const ret, const NODE *c
             const NODE *kw_args = RNODE_HASH(RNODE_HSHPTN(node)->nd_pkwargs)->nd_head;
             keys = rb_ary_new_capa(kw_args ? RNODE_LIST(kw_args)->as.nd_alen/2 : 0);
             while (kw_args) {
-                rb_ary_push(keys, get_symbol_value(RNODE_LIST(kw_args)->nd_head));
+                rb_ary_push(keys, get_symbol_value(iseq, RNODE_LIST(kw_args)->nd_head));
                 kw_args = RNODE_LIST(RNODE_LIST(kw_args)->nd_next)->nd_next;
             }
         }
@@ -7143,7 +7143,7 @@ iseq_compile_pattern_each(rb_iseq_t *iseq, LINK_ANCHOR *const ret, const NODE *c
                 for (i = 0; i < keys_num; i++) {
                     NODE *key_node = RNODE_LIST(args)->nd_head;
                     NODE *value_node = RNODE_LIST(RNODE_LIST(args)->nd_next)->nd_head;
-                    VALUE key = get_symbol_value(key_node);
+                    VALUE key = get_symbol_value(iseq, key_node);

                     ADD_INSN(ret, line_node, dup);
                     ADD_INSN1(ret, line_node, putobject, key);
```

Reply:
Updated to use `UNKNOWN_NODE`.